File: Data Structures & Algorithms/three-integer-sum/submission-3.java
class Solution {
    public static List<List<Integer>> threeSum(int[] nums) {
      List<List<Integer>> res = new ArrayList<>();
      Arrays.sort(nums);

      int i1, i2, i3;
      for (i1 = 0; i1 < nums.length - 2; i1++) {
         if (i1 > 0 && nums[i1] == nums[i1 - 1])
            continue;

         i2 = i1 + 1;
         i3 = nums.length - 1;

         while (i2 < i3) {
            int sum = nums[i1] + nums[i2] + nums[i3];
            if (sum == 0) {
               res.add(Arrays.asList(nums[i1], nums[i2], nums[i3]));
            }

            if (sum < 0) {
               i2++;
               while (i2 < nums.length && nums[i2] == nums[i2 - 1]) {
                  i2++;
               }
            } else {
               i3--;
               while (i3 > 0 && nums[i3] == nums[i3 + 1])
                  i3--;
            }
         }
      }

      return res;
   }
}

File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l = 0
        r = len(nums) - 1

        while l < r:
            if nums[l] != val:
                l += 1
                continue
            if nums[r] == val:
                r -= 1
                continue
            
            nums[l] = nums[r]
            nums[r] = val
            l += 1
            r -= 1

        # return l + 1
        return r - l